### PR TITLE
Relax dependencies on dry-* gems

### PR DESCRIPTION
**Pull request by jwoudenberg**
## Summary

Hi! First of all, thank you for your work on this repo, it's super appreciated!

We've been updating some of our dependencies and found that the current constraints for the `dry-auto_inject` and `dry-configurable` gems are stricter then they need to be. Both gems recently released `1.x` versions, and those to seem to work great. This PR relaxes those constraints to make warden-jwt_auth compatible with these newer versions.

## Checklist

Check out our [PR guidelines](https://github.com/solidusio/.github/blob/master/CONTRIBUTING.md#pull-request-guidelines) for more details.

The following are mandatory for all PRs:

- [x] I have written a thorough PR description.
- [x] I have kept my commits small and atomic.
- [x] I have used clear, explanatory commit messages.

The following are not always needed (~cross them out~ if they are not):

- [ ] ~I have added automated tests to cover my changes.~
- [ ] ~I have attached screenshots to demo visual changes.~
- [ ] ~I have opened a PR to update the [guides](https://github.com/solidusio/edgeguides).~
- [ ] ~I have updated the README to account for my changes.~

### CODE REVIEW

**Comment by thomasdarde** [comment]
This is a duplicate of #46 which also add zeitwerk loading

File: lib/warden/jwt_auth/version.rb [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 module Warden
   module JWTAuth
-    VERSION = '0.7.0'
+    VERSION = '0.7.1'
```
Comment (waiting-for-dev) [inline lib/warden/jwt_auth/version.rb:5]: Can we revert this change? It's a change not related to what the commit describes.

**Review by waiting-for-dev — CHANGES_REQUESTED** [review]

**Comment by waiting-for-dev** [comment]
Many thanks for your collaboration. I'd just to bump the version after merging this.

File: lib/warden/jwt_auth/version.rb (hunk 1 above)
Comment (jwoudenberg) [reply lib/warden/jwt_auth/version.rb]: Good point. Fixed!

**Comment by jwoudenberg** [comment]
> Many thanks for your collaboration. I'd just to bump the version after merging this.

Thank you for the quick response!

**Review by waiting-for-dev — APPROVED** [review]
